mozetl/topline/historical_backfill.py

```diff
@@ -0,0 +1,119 @@
+"""
+# Topline Historical Backfill
+
+This script provides a way to backfill a summmary period from
+historical data generated by the executive_reporting pipeline
+[1]. This data is stored in two files representing the monthly and
+weekly topline summaries. The reporting pipeline has been rewritten in
+`telemetry-batch-view` as the ToplineSummaryView, which writes out
+data to `telemetry-parquet/topline_summary/v1`.
+
+Example Usage:
+
+```
+python -m mozetl.topline.historical_backfill \
+    s3://path/to/v4_weekly.csv \
+    weekly \
+    net-mozaws-prod-us-west-2-pipeline-analysis \
+    --prefix test-topline/topline_summary/v1
+```
+
+[1] https://git.io/v9mxF
+"""
+import json
+import logging
+import os
+
+import click
+
+from pyspark.sql import SparkSession, functions as F
+from pyspark.sql.types import StructType
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def schema_from_json(path, relative=True):
+    """ Create a pyspark schema from the json representation.
+
+    The json representation must be from a StructType. This can be
+    generated from any StructType using the `.json()` method. The
+    schema for a dataframe can be obtained using the `.schema`
+    accessor. For example, to generate the json from the
+    `topline_summary`, run the following in the pyspark repl:
+
+    >>> path = 's3a://telemetry-parquet/topline_summary/v1/mode=weekly'
+    >>> json_data = spark.read.parquet(path).schema.json()
+
+    :path str: Path the the json data
+    :relative bool: Use the relative path to the current file.
+    """
+
+    if relative:
+        path = os.path.join(os.path.dirname(__file__), path)
+
+    with open(path) as json_data:
+        data = json.load(json_data)
+    return StructType.fromJson(data)
+
+
+# Generate module level schemas
+historical_schema = schema_from_json('historical_schema.json')
+topline_schema = schema_from_json('topline_schema.json')
+
+
+def create_s3_path(bucket, prefix, protocol="s3"):
+    return "{}://{}/{}".format(bucket, prefix, protocol)
```

Comment:
I removed the protocol argument because it ends up not being used. I meant for the protocol flag to be set during mocking, but problems with recursive mocking lead to re-implement the function during mocking. 

This is a candidate for refactoring into a separate utility library.